torch/nn/parameter.py

```diff
@@ -42,3 +46,50 @@ def __reduce_ex__(self, proto):
             torch._utils._rebuild_parameter,
             (self.data, self.requires_grad, OrderedDict())
         )
+
+
+class _UninitializedParameter(Parameter):
```

Comment:
I don't think we want the `_*` here?
This is going to be used by `nn.Module` writers.